lib/rules/object-shorthand.js

```diff
@@ -284,20 +284,20 @@ module.exports = {
             const arrowToken = sourceCode.getTokenBefore(node.value.body, astUtils.isArrowToken);
             const fnBody = sourceCode.text.slice(arrowToken.range[1], node.value.range[1]);
 
-            let shouldAddParensAroundParameters = false;
             let tokenBeforeParams;
+            let shouldAddParensAroundParameters = false;
 
-            if (node.value.params.length === 0) {
-                tokenBeforeParams = sourceCode.getFirstToken(node.value, astUtils.isOpeningParenToken);
+            // First token should not be `async`
+            if (node.value.async) {
+                tokenBeforeParams = sourceCode.getFirstToken(node.value, token => !(token.type === "Identifier" && token.value === "async"));
             } else {
-                tokenBeforeParams = sourceCode.getTokenBefore(node.value.params[0]);
+                tokenBeforeParams = sourceCode.getFirstToken(node.value);
             }
 
             if (node.value.params.length === 1) {
-                const hasParen = astUtils.isOpeningParenToken(tokenBeforeParams);
-                const isTokenOutsideNode = tokenBeforeParams.range[0] < node.range[0];
-
-                shouldAddParensAroundParameters = !hasParen || isTokenOutsideNode;
+                shouldAddParensAroundParameters = !astUtils.isOpeningParenToken(
+                    sourceCode.getTokenBefore(node.value.params[0])
+                );
             }
```

Comment:
The range check was intended to determine whether the opening paren is a paren of params or a grouping paren around the key.

Now, for example, if the code looks like this:

```js
const obj = {
    method: (foo => {})
};
```

The autofix would be wrong:

```js
const obj = {
    methodfoo {}
};

```

Apparently, a test case that would have caught this was missing.

Reply:
And the original code wasn't good, it should have been `< node.value.range[0]`, not `< node.range[0]`.

Reply:
I got what's your mean.